### PR TITLE
Allow middle-clicking beatmap scoreboard rows

### LINKED ISSUE #6829: Clicking a score row to open in new tab (middle click or ctrl-click) doesn't behave as expected
when holding the command key on macos and clicking any row in the highscores of any beatmap ([example for convenience](https://osu.ppy.sh/beatmapsets/409164#osu/892780)), the score does not open in a new tab as expected but instead loads in the current tab.

Occurs in both safari and chrome.

### PR DESCRIPTION
**Pull request by notbakaneko**
Makes (almost) the entire score row a link to the score.

For table rows, this means wrapping the contents of each cell in an actual `a` tag; handling middle-click behaviour is a _browser_ feature, it's not triggerable via javascript.

The top and own score portions just have a link element under the block and selectively overrides `pointer-events`

closes #6829

### CODE REVIEW

File: resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee [hunk 1]
```diff
@@ -53,34 +56,49 @@ export class ScoreboardTableRow extends React.PureComponent
...
       for stat in @props.hitTypeMapping
-        td
+        el @tdLink,
           key: stat[0]
-          className: osu.classWithModifiers(cell, ['zero'] if score.statistics["count_#{stat[1]}"] == 0),
+          modifiers: ['zero'] if score.statistics["count_#{stat[1]}"] == 0
           osu.formatNumber(score.statistics["count_#{stat[1]}"])
 
-      td className: osu.classWithModifiers(cell, ['zero'] if score.statistics.count_miss == 0),
+      el @tdLink,
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee:74]: previously the other zeroes turn white as well on hover :thinking:

File: resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee [hunk 2]
```diff
@@ -53,34 +56,49 @@ export class ScoreboardTableRow extends React.PureComponent
...
+        round score.pp
 
-      td className: osu.classWithModifiers(cell, ['time']),
+      el @tdLink,
+        modifiers: ['time']
         el ScoreboardTime,
           dateTime: score.created_at
 
-      td className: osu.classWithModifiers(cell, ['mods']),
+      el @tdLink,
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee:87]: doesn't quite work at all for this column

File: resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee [hunk 3]
```diff
@@ -53,34 +56,49 @@ export class ScoreboardTableRow extends React.PureComponent
 
       td className: cell,
         a
-          className: "#{bn}__user-link js-usercard"
+          className: "#{bn}__cell-content #{bn}__cell-content--user-link js-usercard"
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee:59]: the whole column now links to user. Not sure which one is better.

File: resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee [hunk 4]
```diff
@@ -23,26 +24,28 @@ export class ScoreboardTableRow extends React.PureComponent
...
 
-      td className: osu.classWithModifiers(cell, ['perfect'] if score.accuracy == 1),
+      el @tdLink,
+        modifiers: ['perfect'] if score.accuracy == 1
         "#{osu.formatNumber(score.accuracy * 100, 2)}%"
 
       td className: cell,
         if score.user.country_code
           a
+            className: "#{bn}__cell-content"
```
Comment (nanaya) [inline resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee:48]: there's a blank space still linking to country performance page

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee (hunk 3 above)
Comment (notbakaneko) [reply resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee:59]: The top score one already behaves this way, this makes it behave the same way.

File: resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee (hunk 4 above)
Comment (notbakaneko) [reply resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee:48]: Not sure if removing it or making it dead space is an improvement

Comment (nanaya) [reply resources/assets/coffee/react/beatmapset-page/scoreboard-table-row.coffee:48]: there's hover text on the flag itself so outside of it it shouldn't link to the country perf page anymore

**Review by nanaya — APPROVED** [review]

**Pull request merged**